Automattic/wp-calypso
# Site profiler: Allow providing domain via URL query param

Closes https://github.com/Automattic/wp-calypso/issues/81949

## Proposed Changes

* Introduced a single source of truth for getting the domain from the URL query param.
* Create a hook for getting a domain value. The hook contains logic for the sanitization domain.
* Form submission updates queryParam instead of setting domain field.
* Allowed empty form submission, which resets the app state and shows the initial screen

## Testing Instructions

* Go to `/site-profiler?domain={ANY_DOMAIN}` (provide domain query param)
* Check if it loads proper data
* Update input field with another domain
* Check if the query param is updated

<img width="1416" alt="Screenshot 2023-09-20 at 21 18 04" src="https://github.com/Automattic/wp-calypso/assets/1241413/7cab54d7-de00-40b2-97f3-8bdde60accb1">


**NOTE:**
- Form validation is still missing and will be covered in the future PRs

## Pre-merge Checklist
